app/src/components/pages/collectionsPage/collectionsPage.tsx

```diff
@@ -1,17 +1,113 @@
+import { useSearchParams, usePathname, useRouter } from "next/navigation";
 import { headerBreakpoints } from "../../../utils/breakpoints";
+import { CardsGrid } from "../../grids/cardsGrid";
+import LaneLoading from "../../lane/laneLoading";
+import { displayResults, totalNumPages } from "../../../utils/utils";
+import type { SyntheticEvent } from "react";
 import { createAdobeAnalyticsPageName } from "@/src/utils/utils";
 
 export const CollectionsPage = ({ data }) => {
+  const { push } = useRouter();
+  const searchParams = useSearchParams();
+  const pathname = usePathname();
+  const [isLoaded, setIsLoaded] = useState(false);
+
+  const collections = data.collection; // typeof data.collection === "object" ? [data.collection] : data.collection // this breaks empty search
+  const headingRef = useRef<HTMLHeadingElement>(null);
+
+  // pagination
+  const numFound = data.numFound ? data.numFound : data.numResults;
+  const totalPages = totalNumPages(numFound, data.perPage);
+
+  // defaults
+  let currentPage = Number(searchParams.get("page")) || 1;
+  let currentSort = searchParams.get("sort") || "date-desc";
+  let currentCollectionKeyword = searchParams.get("collection_keyword") || "";
+
+  const handleSearchSubmit = async (e: SyntheticEvent) => {
+    e.preventDefault();
+    const queryString = createQueryStringFromObject({
+      collection_keyword: currentCollectionKeyword,
+      sort: currentSort,
+      page: currentPage,
+    });
+    setIsLoaded(false);
+    await push(`${pathname}?${queryString}`);
+    setTimeout(() => {
+      setIsLoaded(true);
+      headingRef.current?.focus();
+    }, 2000);
+  };
+
+  // I'm not actually sure if this is necessary but it's in the research catalog
+  const handleSearchChange = (e: SyntheticEvent) => {
+    const target = e.target as HTMLInputElement;
+    currentCollectionKeyword = target.value;
+  };
+
+  // pagination
+  // Question: Do we want to introduce debouncing?
+  const onPageChange = async (pageNumber: number) => {
+    currentPage = pageNumber;
+    const queryString = createQueryStringFromObject({
+      collection_keyword: currentCollectionKeyword,
+      sort: currentSort,
+      page: pageNumber.toString(),
+    });
+    setIsLoaded(false);
+    await push(`${pathname}?${queryString}`);
+    setTimeout(() => {
+      setIsLoaded(true);
+      headingRef.current?.focus();
+    }, 2000);
+  };
+
+  // TODO: right now, whenever a user hits earch, the url is created with all params including the defaults.
+  // Perhaps the query string should only include the values that are not defaults.
+  const createQueryStringFromObject = (paramObj) => {
+    const params = new URLSearchParams();
+    Object.keys(paramObj).map((name, value) => {
+      params.set(
+        name.toString(),
+        name === "page" ? paramObj[name].toString() : paramObj[name]
+      );
+    });
+    return params.toString();
+  };
+
+  // sort
+  const onMenuClick = async (id) => {
```

Comment:
When I test this, "Z to A" returns results in A to Z and "A to Z" returns results in Z to A. So the ordering is off somewhere.

Reply:
Ah this must have something to do with my misunderstanding of the concept of title "ascending" or "descending."

Reply:
try again?

Reply:
Works fine.